app/ore/project/io/PluginFileData.scala

```diff
@@ -0,0 +1,216 @@
+package ore.project.io
+
+import java.io.BufferedReader
+
+import com.google.gson.JsonParser
+import com.google.gson.stream.JsonReader
+import models.project.{Tag, TagColors}
+import ore.project.Dependency
+
+import scala.collection.mutable.ListBuffer
+
+/**
+  * The metadata within a [[PluginFile]]
+  *
+  * @author phase
+  * @param data the data within a [[PluginFile]]
+  */
+class PluginFileData(var data: List[DataValue[_]]) {
+
+  data = data.groupBy(_.key).flatMap { case (key, value) =>
```

Comment:
Can we make a factory that does this instead and get away with a val instead of var?